crates/orchestrator/src/tests/server/job_routes.rs

```diff
@@ -0,0 +1,102 @@
+use std::io::Read;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use axum::http::StatusCode;
+use chrono::{SubsecRound as _, Utc};
+use hyper::body::Buf;
+use hyper::{Body, Request};
+use rstest::*;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use url::Url;
+use utils::env_utils::get_env_var_or_default;
+use uuid::Uuid;
+
+use crate::config::{init_config, Config};
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::queue::init_consumers;
+use crate::routes::job_routes::job_routes;
+use crate::tests::config::TestConfigBuilder;
+
+#[fixture]
+pub async fn setup_job_server() -> (SocketAddr, Arc<Config>) {
+    let provider = JsonRpcClient::new(HttpTransport::new(
+        Url::parse("http://localhost:9944".to_string().as_str()).expect("Failed to parse URL"),
+    ));
+
+    TestConfigBuilder::new().configure_starknet_client(provider.into()).build().await;
+
+    let host = get_env_var_or_default("HOST", "127.0.0.1");
+    let port = get_env_var_or_default("PORT", "3000").parse::<u16>().expect("PORT must be a u16");
+    let address = format!("{}:{}", host, port);
+
+    let listener = tokio::net::TcpListener::bind(address.clone()).await.expect("Failed to get listener");
+    let addr = listener.local_addr().unwrap();
```

Comment:
remove unwrap

Reply:
I personally think `unwraps` in `tests` are not a problem, currently changing it to `expect` for a better error message, lemme know what you think @Mohiiit

Reply:
yes true that I think, you can ignore for these changes